WordPress/Classes/ViewRelated/Post/PostEditor+MoreOptions.swift

```diff
@@ -22,10 +22,28 @@ extension PostEditor where Self: UIViewController {
 
     func displayHistory() {
         let revisionsViewController = RevisionsTableViewController(post: post) { [weak self] revision in
-            if let post = self?.post.update(from: revision) {
-                DispatchQueue.main.async {
-                    self?.post = post
+            guard let self = self else {
+                return
+            }
+            let post = self.post.update(from: revision)
+
+            // show the notice with undo button
+            let notice = Notice(title: "Revision loaded", message: nil, feedbackType: .success, notificationInfo: nil, actionTitle: "Undo", cancelTitle: nil) { [weak self] (happened) in
+                guard happened, let self = self else {
+                    return
                 }
+                DispatchQueue.main.async { [weak self] in
+                    guard let self = self, let original = self.post.original else {
+                        return
+                    }
+                    let clone = self.post.clone(from: original)
+                    self.post = clone
```

Comment:
Even here, no weak self needed and you can `guard let original` and `let clone`. `self.post` can also remain optional.